doc/src/manual/performance-tips.md

```diff
@@ -48,79 +54,79 @@ so all the performance issues discussed previously apply.
 
 ## Measure performance with [`@time`](@ref) and pay attention to memory allocation
 
-A useful tool for measuring performance is the [`@time`](@ref) macro. The following example
-illustrates good working style:
+A useful tool for measuring performance is the [`@time`](@ref) macro. We here repeat the example
+with the global variable above, but this time with the type annotation removed:
 
-```julia-repl
-julia> function f(n)
-           s = 0
-           for i = 1:n
-               s += i/2
+```jldoctest; setup = :(using Random; srand(1234)), filter = r"[0-9\.]+ seconds \(.*?\)"
+julia> x = rand(1000);
+
+julia> function sum_global()
+           s = 0.0
+           for i in x
+               s += i
            end
-           s
-       end
-f (generic function with 1 method)
+           return s
+       end;
 
-julia> @time f(1)
-  0.012686 seconds (2.09 k allocations: 103.421 KiB)
-0.5
+julia> @time sum_global()
+  0.008089 seconds (4.17 k allocations: 106.747 KiB)
+496.84883432553846
 
-julia> @time f(10^6)
-  0.021061 seconds (3.00 M allocations: 45.777 MiB, 11.69% gc time)
-2.5000025e11
+julia> @time sum_global()
+  0.000140 seconds (3.49 k allocations: 70.313 KiB)
+496.84883432553846
 ```
 
-On the first call (`@time f(1)`), `f` gets compiled.  (If you've not yet used [`@time`](@ref)
+On the first call (`@time sum_global()`) the function gets compiled. (If you've not yet used [`@time`](@ref)
 in this session, it will also compile functions needed for timing.)  You should not take the results
 of this run seriously. For the second run, note that in addition to reporting the time, it also
-indicated that a large amount of memory was allocated.
+indicated that a significant amount of memory was allocated. We are here just computing a sum over all elements in
+a vector of 64-bit floats so there should be no need to allocate memory (at least not on the heap which is what `@time` reports).
 
 Unexpected memory allocation is almost always a sign of some problem with your code, usually a
 problem with type-stability. Consequently, in addition to the allocation itself, it's very likely
 that the code generated for your function is far from optimal. Take such indications seriously
 and follow the advice below.
 
-For more serious benchmarking, consider the [BenchmarkTools.jl](https://github.com/JuliaCI/BenchmarkTools.jl)
-package which evaluates the function multiple times in order to reduce noise.
 
-As a teaser, an improved version of this function allocates no memory
+If we instead pass `x` as an argument to the function it no longer allocates memory
 (the allocation reported below is due to running the `@time` macro in global scope)
-and has an order of magnitude faster execution after the first call:
+and is significantly faster after the first call:
 
 ```julia-repl
-julia> @time f_improved(1)
-  0.007008 seconds (1.32 k allocations: 63.640 KiB)
-0.5
+julia> @time sum_arg(x)
```

Comment:
`sum_arg` hasn't been defined anywhere?

Reply:
Neither was `f_improved`. I think the point before was to not define it to spur the readers interest but it is a bit more obvious what the problem is now since it has been discussed earlier.

Reply:
But why not define `sum_arg(x)` just like we defined `sum_global` above?

Reply:
Did that